src/improvements/README.md

```diff
@@ -1,59 +1,200 @@
+   - Network-specific configurations
 
-### L2Chains
+## Quick Start
 
+1. Create a new task:
+```bash
+cd src/improvements/
+just new task
+```
+
+2. Configure the task in `config.toml`:
 ```toml
-# L2Chains is a list of the L2 chains that the task will interact with
-l2chains = [{"name": "Base", "chainId": 8453}, {"name": "Metal", "chainId": 1750}, {"name": "OP Mainnet", "chainId": 10}]
+templateName = "GasConfigTemplate"
+l2chains = [{"name": "OP Mainnet", "chainId": 10}]
 ```
 
-### Task Template Files
+3. Test the task:
+```bash
+forge script <template-path> --sig "run(string)" <config-path> --rpc-url devnet -vvv
+```
 
-Task templates allow task developers to create new tasks without writing any Solidity code. The templates are designed to be as flexible as possible, allowing developers to create tasks that update multiple parameters for a single or superchain. Templates are currently in development and will be released in a future PR.
+## Available Templates
 
-The mainnet configuration file for task template example 00 can be found here: [task-00/mainnetConfig.toml](./example/task-00/mainnetConfig.toml)
+1. **Gas Config Template**
+   - Set gas limits for L2 chains
+   - Configure transaction parameters
+   - Example: [Gas Config Example](#gas-config-example)
 
-```toml
-[gasConfigs]
-gasLimits = [{chainId = 291, gasLimit = 100000000}, {chainId = 1750, gasLimit = 100000000}]
-```
+2. **Dispute Game Template**
+   - Configure dispute game implementations
+   - Set game parameters
+   - Example: [Dispute Game Example](#dispute-game-example)
 
-This toml configuration file allows task developers to set gas limits for the task. After the changes are applied in the simulation, validations are run to ensure that the new values are the expected values.
+3. **Game Type Template**
+   - Set respected game types
+   - Configure chain permissions
+   - Example: [Game Type Example](#game-type-example)
 
-### Running Example Tasks
+## Documentation
 
-#### Template 00 to set gas configs:
+### Core Concepts
+- [Template Architecture Guide](./doc/TEMPLATE_ARCHITECTURE.md)
+  * Template configuration structure
+  * TOML file architecture
+  * Task status definitions
+  * Validation system
 
-```bash
-forge script src/improvements/template/GasConfigTemplate.sol --sig "run(string)" test/tasks/mock/example/task-00/config.toml --rpc-url mainnet -vvv
-```
+- [Task Creation Guide](./doc/TASK_CREATION_GUIDE.md)
+  * Step-by-step task creation
+  * Using the justfile scaffolding
+  * Best practices
+  * Troubleshooting
 
-#### Template 01 to set dispute game upgrade:
+- [New Template Guide](./doc/NEW_TEMPLATE_GUIDE.md)
+  * Creating Solidity templates
+  * Required implementations
+  * Testing requirements
+  * Example walkthrough
 
-```bash
-forge script src/improvements/template/DisputeGameUpgradeTemplate.sol --sig "run(string)" test/tasks/mock/example/task-01/config.toml --rpc-url mainnet -vvv
-```
+## Example Configurations
```

Comment:
I would have a max of one here then link to example configs. e.g. keep Gas Congif Example and drop the rest.